src/Agent.Worker/Build/BuildServer.cs

```diff
@@ -34,6 +35,7 @@ public BuildServer(VssConnection connection, Guid projectId)
             Build2.BuildArtifact artifact = new Build2.BuildArtifact()
             {
                 Name = name,
+                Source = jobId,
```

Comment:
why this is not a property on the `propertiesDictionary`?

Reply:
we want first class support for source, so that we can index etc., so this part of artifact contract it'self and not just some prop for resource.

Reply:
reference: [azuredevops PR 470152](https://dev.azure.com/mseng/_git/AzureDevOps/pullrequest/470152?_a=overview)